src/devices/Ft4222/FtFunction.cs

```diff
@@ -12,15 +12,133 @@ namespace Iot.Device.Ft4222
         /// </summary>
         /// <param name="numdevs">number of devices</param>
         /// <returns>The status</returns>
-        [DllImport("ftd2xx")]
-        public static extern FtStatus FT_CreateDeviceInfoList(out uint numdevs);
+        public static FtStatus FT_CreateDeviceInfoList(out uint numdevs)
+        {
+            if (Environment.OSVersion.Platform == PlatformID.Unix)
+            {
+                return FT_CreateDeviceInfoListLinux(out numdevs);
+            }
+
+            return FT_CreateDeviceInfoListWin(out numdevs);
+        }
+
+        /// <summary>
+        /// Get Device Information Detail
+        /// </summary>
+        /// <param name="index">Index of the device</param>
+        /// <param name="flags">Flags</param>
+        /// <param name="chiptype">Device type</param>
+        /// <param name="id">ID</param>
+        /// <param name="locid">Location ID</param>
+        /// <param name="serialnumber">Serial Number</param>
+        /// <param name="description">Description</param>
+        /// <param name="ftHandle">Handle</param>
+        /// <returns>The status</returns>
+        public static FtStatus FT_GetDeviceInfoDetail(uint index, out uint flags, out FtDevice chiptype, out uint id, out uint locid, in byte serialnumber, in byte description, out IntPtr ftHandle)
+        {
+            if (Environment.OSVersion.Platform == PlatformID.Unix)
+            {
+                return FT_GetDeviceInfoDetailLinux(index, out flags, out chiptype, out id, out locid, in serialnumber, in description, out ftHandle);
+            }
+
+            return FT_GetDeviceInfoDetailWin(index, out flags, out chiptype, out id, out locid, in serialnumber, in description, out ftHandle);
+        }
+
+        /// <summary>
+        /// Open a device
+        /// </summary>
+        /// <param name="pvArg1">The device element identifying the device, depends on the flag</param>
+        /// <param name="dwFlags">The flag how to open the device</param>
+        /// <param name="ftHandle">The handle of the open device</param>
+        /// <returns>The status</returns>
+        public static FtStatus FT_OpenEx(uint pvArg1, FtOpenType dwFlags, out SafeFtHandle ftHandle)
+        {
+            if (Environment.OSVersion.Platform == PlatformID.Unix)
+            {
+                return FT_OpenExLinux(pvArg1, dwFlags, out ftHandle);
+            }
+
+            return FT_OpenExWin(pvArg1, dwFlags, out ftHandle);
+        }
+
+        /// <summary>
+        /// Close the device
+        /// </summary>
+        /// <param name="ftHandle">The device handle</param>
+        /// <returns>The status</returns>
+        public static FtStatus FT_Close(SafeFtHandle ftHandle)
+        {
+            if (Environment.OSVersion.Platform == PlatformID.Unix)
+            {
+                return FT_CloseLinux(ftHandle);
+            }
+
+            return FT_CloseWin(ftHandle);
+        }
+
+        #endregion
+
+        #region Linux ftd2xx
```

Comment:
We usually try to keep the Interop methods on their own file

Reply:
You mean doing 2 different files? In this case, it's just because on Linux, the drivers are statically linked to the ft4222 library while on Windows (and Mac) they are in 2 different libraries.

Reply:
I meant doing it similar to System.Device.Gpio where we have special interop files for anything that will PInvoke, and the managed code is located on its own class. That said I don't have a big push on this since this is a device binding so we are less strict about this here, so I'm fine with leaving as is.